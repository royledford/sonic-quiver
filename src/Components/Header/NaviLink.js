import React, { Component } from 'react'
import { NavLink } from 'react-router-dom'

import PropTypes from 'prop-types'
import './NaviLink.css'

export default class NaviLink extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    linkText: PropTypes.string.isRequired,
    largeText: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    activeClassName: PropTypes.string,
  }
  static defaultProps = {
    activeClassName: '',
  }

  render() {
    const { to, id, linkText, largeText, description, activeClassName } = this.props
    return (
      <div className="navilink-wrap">
        <NavLink to={to} id={id} className="navilink-link" activeClassName={activeClassName}>
          {linkText}
        </NavLink>
        <div id={`{id}-largeText`} className="navilink-largeText">
          {largeText}
        </div>
        <div id={`{id}-description`} className="navilink-description">
          {description}
        </div>
      </div>
    )
  }
}
