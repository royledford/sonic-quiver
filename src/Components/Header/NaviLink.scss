@import 'styles/vars.scss';

.navilink-wrap {
  position: relative;
  display: inline-block;
  height: 28px;
}

.navilink-link {
  position: relative;

  padding: 10px 50px 10px 50px;

  color: white;
  font-size: 12px;
  white-space: nowrap;
  letter-spacing: 2px;
  text-decoration: none;
  z-index: 3;
}

.navilink-largeText {
  position: absolute;
  display: none;
  top: 0px;
  left: -250px;

  color: $link-largetext;
  font-family: 'Montserrat', sans-serif;
  font-size: 80px;
  font-weight: bold;
  font-weight: 900;

  opacity: 0;
  z-index: 1;
  pointer-events: none;
}

.navilink-description {
  display: none;
  position: absolute;
  top: 35px;
  left: 200px;
  width: 300px;

  color: $link-smalltext;
  white-space: nowrap;
  font-size: 14px;
  text-align: center;

  pointer-events: none;
  opacity: 0;
  z-index: 2;
}

.navilink-link:hover ~ .navilink-largeText {
  display: inline-block;
  animation: showLinkText 1000ms ease-in-out forwards;
}
.navilink-link:hover ~ .navilink-description {
  display: inline-block;
  animation: showLinkText 600ms ease-in-out forwards;
}

.navilink-link:visited {
  color: white;
}
.navilink-link:hover {
  color: white;
  font-weight: bold;
}
.navilink-link:active {
  color: white;
}


@keyframes showLinkText {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
    left: -50px;
  }
}

