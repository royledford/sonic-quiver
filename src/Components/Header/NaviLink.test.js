import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import NaviLink from './NaviLink'
import renderer from 'react-test-renderer'

it('renders', () => {
  const tree = renderer
    .create(
      <Router>
        <NaviLink
          to="http:\\www.royledford.com"
          id="ElementId"
          linkText="Test Link"
          largeText="Big Text"
          description="test description"
        />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
