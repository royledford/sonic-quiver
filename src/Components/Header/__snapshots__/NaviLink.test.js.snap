// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="navilink-wrap"
>
  <a
    aria-current={false}
    className="navilink-link"
    href="http:\\\\\\\\www.royledford.com"
    id="ElementId"
    onClick={[Function]}
    style={undefined}
  >
    Test Link
  </a>
  <div
    className="navilink-largeText"
    id="{id}-largeText"
  >
    Big Text
  </div>
  <div
    className="navilink-description"
    id="{id}-description"
  >
    test description
  </div>
</div>
`;
